scripts/compile_cost_assumptions.py

```diff
@@ -1574,6 +1593,214 @@ def add_mean_solar_rooftop(data):
+            df.loc[charger_investment_filter & df.year==2030, "value"] += agg.loc[(tech, 2030)]/2
+            df.loc[discharger_investment_filter & df.year==2021, "value"] += agg.loc[(tech, 2021)]/2
+            df.loc[discharger_investment_filter & df.year==2030, "value"] += agg.loc[(tech, 2030)]/2
+    df.loc[:,"technology"] = df["technology"] + "-" + df["technology_type"]
+
+    # aggregate technology_type and unit
+    df = df.groupby(["technology", "unit", "year"]).agg({
+        'technology': 'first',
+        'year': 'first',
+        'parameter': 'first',
+        'value': 'sum',
+        'unit': 'first',
+        'type': 'first',
+        'carrier': 'first',
+        'technology_type': 'first',
+        'source': 'first',
+        'note': 'first',
+        'reference': 'first',
+    }).reset_index(drop=True)
+
+    # calculate %/year FOM on aggregated values
+    for tech in df.technology.unique():
+        for year in df.year.unique():
+            df_tech = df.loc[(df.technology == tech) & (df.year == year)].copy()
+            a = df_tech.loc[df_tech.unit=="EUR/MW-year", "value"].values
+            b = df_tech.loc[df_tech.unit=="EUR/MW", "value"].values
+            df.loc[df_tech.loc[df_tech.unit=="EUR/MW-year"].index, "value"] = a / b * 100 # EUR/MW-year / EUR/MW = %/year
+            c = df_tech.loc[df_tech.unit=="EUR/MWh-year", "value"].values
+            d = df_tech.loc[df_tech.unit=="EUR/MWh", "value"].values
+            df.loc[df_tech.loc[df_tech.unit=="EUR/MWh-year"].index, "value"] = c / d * 100 # EUR/MWh-year / EUR/MWh = %/year
+
+    df.loc[:,"unit"] = df.unit.str.replace("EUR/MW-year", "%/year")
+    df.loc[:,"unit"] = df.unit.str.replace("EUR/MWh-year", "%/year")
+
+    # c) Linear Inter/Extrapolation
+    # data available for 2021 and 2030, but value for given "year" passed by function needs to be calculated
+    for tech in df.technology.unique():
+        for param in df.parameter.unique():
+            filter = (df.technology == tech) & (df.parameter == param)
+            y = df.loc[filter, "value"]
+            if y.empty:
+                continue  # nothing to interpolate
+            elif y.iloc[0]==y.iloc[1] or param=="efficiency" or param=="lifetime":
+                ynew = y.iloc[1]  # assume new value is the same as 2030
+            elif y.iloc[0]!=y.iloc[1]:
+                x = df.loc[filter, "year"] # both values 2021+2030
+                first_segment_diff = y.iloc[0]-y.iloc[1]
+                endp_first_segment = y.iloc[1]
+                # Below we create linear segments between 2021-2030
+                # While the first segment is known, the others are defined by the initial segments with a accumulating quadratic descreasing gradient
+                other_segments_points = [2034, 2039, 2044, 2049, 2054, 2059]
+                
+                def geometric_series(nominator, denominator=1, number_of_terms=1, start=1):
```

Comment:
Can you change this, to either find cost assumptions for 2050 or assuming the same cost assumptions for the following years from 2030 onwards?

Reply:
The config.yaml now has the option `approx_beyond_2030` with two options `"geometric_series"` (optimistic that technologies will improve) and `"same_as_2030"` (very pessimistic assuming costs are not decreasing after 2030). Below I am also just showing briefly that the options work fine.

![image](https://github.com/PyPSA/technology-data/assets/61968949/49396235-cc69-4796-8620-4ee77826eb47)

# Option "same_as_2030"
## 2020 results
![image](https://github.com/PyPSA/technology-data/assets/61968949/19b3d2b7-cac3-484b-9d50-5c17e87d698d)
## 2030 results
![image](https://github.com/PyPSA/technology-data/assets/61968949/10e86959-79b4-4aca-b8df-ff0220ece85b)
## 2050 results
![image](https://github.com/PyPSA/technology-data/assets/61968949/4c0959fd-21b3-4f9c-b85f-3e5985c0cb43)


# Option "geometric_series"
## 2050 results
![image](https://github.com/PyPSA/technology-data/assets/61968949/f396c967-7207-4aa0-826c-8711821c2e7b)